### ISSUE TITLE
max-statements-per-line conflicting with no-unexpected-multiline 

### ISSUE THREAD

**Issue by vtfn** [eslint/eslint#6153]
With eslint v2.9.0 and this rules

```json
    "max-statements-per-line": [2, {
        "max": 1
      }],
    "no-unexpected-multiline": 2,
```

```max-statements-per-line``` warn about code like this

```javascript
const name = 'ESLint'

;(function foo() {
})()
```

with this message

```
This line has too many statements. Maximum allowed is 1  max-statements-per-line
```
I think ```max-statements-per-line``` should this kind of use of the semicolon.

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed it.
I think `max-statements-per-line` rule should ignore empty statements because empty statements are warned by `no-extra-semi` rule.

**Comment by vtfn** [comment]
Should this be fixed already in 2.10.1?
If so I am still having the same issue.

**Comment by alberto** [comment]
@vtfn oh, I see. I missed the first line and it seems I didn't completely fix the issue.

**Comment by mysticatea** [comment]
Ops, I confirmed it. It's my mistake.

![image](https://cloud.githubusercontent.com/assets/1937871/15269457/c16a21a4-1a39-11e6-8ced-454190d670be.png)

The `const` declaration have that range, so 2 statements exist in line 5.
We should use the last token's position at [max-statements-per-line.js#L78](https://github.com/eslint/eslint/blob/51e05af39829b047cead58c1f89f0f3d48bba7dc/lib/rules/max-statements-per-line.js#L78)

**Comment by alberto** [comment]
@mysticatea I am not sure what you are proposing, or what to do here, since we shouldn't skip every multiline statement. Do you want to take over this?

**Comment by mysticatea** [comment]
Sure, I'm working on this.

**Comment by vtfn** [comment]
Still having the same problem

**Comment by alberto** [comment]
@vtfn this was just merged. did you try it on master?

**Comment by vtfn** [comment]
Yeah, the master worked. I thought that, as the version was 2.10.2 already, it should be working. My mystake, sorry, and thanks for the fix by the way.
